src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -410,7 +411,7 @@ class AppsController(
 
   private def listVersions(appId: PathId)(implicit identity: Identity): Route = {
     val versions = groupManager.appVersions(appId).runWith(Sink.seq)
-    authorized(ViewRunSpec, groupManager.app(appId), Rejections.EntityNotFound.noApp(appId)).apply {
+    authorized(ViewRunSpec, groupManager.runSpec(appId).get, Rejections.EntityNotFound.noApp(appId)).apply {
```

Comment:
has to be on the runspec.. not the app

Reply:
appdefinition is a subtype of runspec so I think we can use appDefinition here...

I will edit this one and commit and you can review if you think it is OK...